spec/support/fetch_secrets_helper.rb

```diff
@@ -16,3 +34,21 @@ def mock_fetch_secrets(is_successful:, fetched_secrets:)
     end
   end
 end
+
+shared_examples_for "it fails when variable is missing or has no value" do |variable_name|
+  it "fails when variable is missing" do
+    allow(Resource).to receive(:[])
+                         .with(/#{account}:variable:conjur\/#{authenticator_name}\/#{service}\/#{variable_name}/)
+                         .and_return(nil)
+
+    expect { subject }.to raise_error(Errors::Conjur::RequiredResourceMissing)
+  end
+
+  it "fails when variable has no value" do
+    allow(Resource).to receive(:[])
```

Comment:
The test here is when the "variable has no value", the `resource_without_value` mock you created expects/can receive a resource so instead of `(:[])`, I would send a value to accurately mock what is happening in reality and it will just return nil anyway

Reply:
This is actually what happens here :)